all-actors/src/main/java/org/sunbird/actors/BaseActor.java

```diff
@@ -1,53 +1,56 @@
-package org.sunbird;
+package org.sunbird.actors;
 
 import akka.actor.UntypedAbstractActor;
 import akka.event.DiagnosticLoggingAdapter;
 import akka.event.Logging;
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Locale;
 import java.util.Map;
-import java.util.UUID;
-import org.sunbird.message.IResponseMessage;
+import org.sunbird.ActorServiceException;
+import org.sunbird.BaseException;
+import org.sunbird.BaseLogger;
 import org.sunbird.message.Localizer;
-import org.sunbird.message.ResponseCode;
 import org.sunbird.request.Request;
+import org.sunbird.response.Response;
 import org.sunbird.util.JsonKey;
 
-/** @author Amit Kumar */
 public abstract class BaseActor extends UntypedAbstractActor {
 
-  public final DiagnosticLoggingAdapter logger = Logging.getLogger(this);
+  final DiagnosticLoggingAdapter logger = Logging.getLogger(this);
 
   public abstract void onReceive(Request request) throws Throwable;
 
   protected Localizer localizer = Localizer.getInstance();
 
   @Override
   public void onReceive(Object message) throws Throwable {
+    Map<String, Object> trace = new HashMap<>();
     if (message instanceof Request) {
       Request request = (Request) message;
       String operation = request.getOperation();
-      Map<String, Object> mdc = new HashMap<>();
-      String requestId = request.getRequestId();
-      if (null == requestId) {
-        requestId = UUID.randomUUID().toString();
+
+      if (request.getHeaders().containsKey(JsonKey.REQUEST_MESSAGE_ID)) {
+        ArrayList<String> requestIds =
```

Comment:
let us add a simple method in Request class that can give us this header. I expect there are multiple places where this is needed and we will repeat this block everywhere.